lib/msf/core/payload/apk.rb

```diff
@@ -295,7 +295,7 @@ def backdoor_apk(apkfile, raw_payload, signature = true, manifest = true, apk_da
     end
 
     print_status "Decompiling original APK..\n"
-    apktool_output = run_cmd(['apktool', 'd', "#{tempdir}/original.apk", '-o', "#{tempdir}/original"])
+    apktool_output = run_cmd(['apktool', 'd', "#{tempdir}/original.apk", '--only-main-classes', '-o', "#{tempdir}/original"])
```

Comment:
Not a blocker, A question that comes to mind is what versions or `apktool` support this flag?

Reply:
The flag has been supported since at least Nov 16, 2019. The flag was available prior, but had some bugs. (https://github.com/iBotPeaches/Apktool/pull/2226)

Version 2.4.1 was released not long after (Nov 30, 2019) and should support this flag:

https://github.com/iBotPeaches/Apktool/releases/tag/v2.4.1

Metasploit requires apktool version 2.0.1 at minimum and raises a warning for version 2.5.1 and prior. 

https://github.com/rapid7/metasploit-framework/blob/fbf2e5d37013badaf0dc98406ad4953ac4d41928/lib/msf/core/payload/apk.rb#L242-L248

This change requires the minimum version to be bumped to 2.4.1.

Reply:
Updated minimum required version to 2.4.1.